libarchive/test/test_write_disk_perms.c

```diff
@@ -216,7 +216,7 @@ DEFINE_TEST(test_write_disk_perms)
 
 	/* Check original owner. */
 	assertEqualInt(0, stat("dir_owner", &st));
-	failure("dir_owner: st.st_uid=%d", st.st_uid);
+	failure("dir_owner: st.st_uid=%jd", (intmax_t)st.st_uid);
```

Comment:
Annoyingly, POSIX states that `uid_t` "shall be integer types".  Unlike some other types mentioned in `<sys/types.h>`, it doesn't specify signed or unsigned.

(I'm guessing that some platforms have signed `uid_t` in order to return -1 from some user-related functions.)

With that in mind, the safest way to print `uid_t` is with `(intmax_t)`.

https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/sys_types.h.html